### PR TITLE
Show outstanding migrations on the error page

### PR DESCRIPTION
**Pull request by hahmed**
Show the outstanding migrations in the UI/console/logs when we have pending migrations, making it easier to know exactly which and how many migrations are pending.

### CODE REVIEW

File: activerecord/lib/active_record/migration.rb [hunk 1]
```diff
@@ -131,15 +131,15 @@ def initialize(name = nil)
...
     include ActiveSupport::ActionableError
 
     action "Run pending migrations" do
       ActiveRecord::Tasks::DatabaseTasks.migrate
 
       if ActiveRecord::Base.dump_schema_after_migration
-        ActiveRecord::Tasks::DatabaseTasks.dump_schema(
-          ActiveRecord::Base.connection_db_config
-        )
+        ActiveRecord::Tasks::DatabaseTasks.dump_schema(ActiveRecord::Base.connection_db_config)
```
Comment (tgxworld) [inline activerecord/lib/active_record/migration.rb:142]: I think we should avoid changing this for cosmetic reasons.

File: activerecord/lib/active_record/migration.rb [hunk 2]
```diff
@@ -148,8 +148,14 @@ def initialize(message = nil)
         super("Migrations are pending. To resolve this issue, run:\n\n        bin/rails db:migrate RAILS_ENV=#{::Rails.env}")
       elsif !message
         super("Migrations are pending. To resolve this issue, run:\n\n        bin/rails db:migrate")
-      else
-        super
+      end
+
+      pending_migrations = ActiveRecord::Base.connection.migration_context.open.pending_migrations
+
+      @details = "You have #{pending_migrations.size} pending #{pending_migrations.size > 1 ? 'migrations:' : 'migration:'}\n\n"
```
Comment (tgxworld) [inline activerecord/lib/active_record/migration.rb:155]: What do you think about including this information in the error message instead? This way, the pending migrations are shown in the logs as well as the UI.

File: activerecord/lib/active_record/migration.rb (hunk 1 above)
Comment (hahmed) [reply activerecord/lib/active_record/migration.rb:142]: Whoops, I will fix this -- thanks!

File: activerecord/lib/active_record/migration.rb (hunk 2 above)
Comment (hahmed) [reply activerecord/lib/active_record/migration.rb:155]: Sure -- I can add the detail to the message attribute, the message attribute is displayed in the log/console.

The reason I ended up splitting out into it's own attribute so we can have a clear title at the top, that is not formatted and a second attribute that adds a more detail which is formatted (it adds new lines automatically) i.e. `<%= simple_format h(@exception.details) %>`

Comment (hahmed) [reply activerecord/lib/active_record/migration.rb:157]: NOTE: once https://github.com/rails/rails/pull/39723 is merged, I will rebase this PR and use the `message` attribute instead.

**Pull request merged**